### ISSUE TITLE
problems using {!join} and {!tag} in filter query

### ISSUE THREAD

**Issue by ssp** [solariumphp/solarium#177]
I am doing somewhat convoluted queries using Solr 4’s {!join …} and also use these for the filter queries for faceting. Using this in a _tagged_ filter query results in an invalid Solr query containing:

``` solr
{!tag=facet-orden}{!join from=kloster_id to=id}(orden_facet:"Trappisten" AND typ:standort-orden)
```

where the code creating it is

``` php
$query->createFilterQuery($queryTag)->setQuery($facetQuery);
```

with `$facetQuery` being

``` solr
{!join from=kloster_id to=id}(orden_facet:"Trappisten" AND typ:standort-orden)
```

and `$queryTag` being

``` php
array('tag' => 'facet-orden', 'key' => 'facet-orden-Trappisten')
```

The error message given by Solr is

``` json
{
  "responseHeader": {
    "status": 400,
    "QTime": 1,
    "params": {
      "indent": "true",
      "q": "*:*",
      "_": "1373374338128",
      "wt": "json",
      "fq": "{!tag=facet-orden}{!join from=kloster_id to=id}(orden_facet:\"Trappisten\" AND typ:standort-orden)"
    }
  },
  "error": {
    "msg": "org.apache.solr.search.SyntaxError: Cannot parse '(orden_facet:\"Trappisten\"': Encountered \"\" at line 1, column 25.\nWas expecting one of:\n     ...\n     ...\n     ...\n    \"+\" ...\n    \"-\" ...\n     ...\n    \"(\" ...\n    \")\" ...\n    \"*\" ...\n    \"^\" ...\n     ...\n     ...\n     ...\n     ...\n     ...\n     ...\n    \"[\" ...\n    \"{\" ...\n     ...\n     ...\n    ",
    "code": 400
  }
}
```

I am using the latest develop version (c379) of Solarium.

**Comment by MadEgg** [comment]
I just ran into the same problem. I am using the setTag() system of the FilterQuery object to set a tag. I would also like to use the edismax-parser for the filter query, so I tried prepending {!type=edismax} to the start of the query. That gives the same error, as the resulting query looks like
{!tag=custom_filter}{!type=edismax} which is not allowed.

As a result I no longer use the setTag method, and add the tag in manually, combined with the type localparam. 

It would be great if more localparams (or even arbitrary localparams) would be supported by the FilterQuery!
